addOns/network/src/main/java/org/zaproxy/addon/network/internal/server/http/handlers/LegacyProxyListenerHandler.java

```diff
@@ -162,7 +162,11 @@ public void handleMessage(HttpMessageHandlerContext ctx, HttpMessage message) {
             }
         }
 
-        synchronized (proxyListeners) {
+        /*
+        It seems that synchronized here is just as general recommendation, not as real necessity.
+        No concurrent changes of state of proxyListeners list, or listeners themselves.
```

Comment:
Just a note for further changes, this assertion is not correct, the `proxyListeners` can be and is modified concurrently so just removing the sync block is not correct (e.g. use a list that supports modifications while being iterated).

Reply:
Instead of synchronizedList would CopyOnWriteArrayList suite ?

Reply:
It does. And I'm fine if that gets in before fixing other parts, even though that changes expectations…